src/protobuf-net.Core/Meta/TypeModel.cs

```diff
@@ -126,7 +126,7 @@ internal static WireType GetWireType(TypeModel model, DataFormat format, Type ty
                 return format == DataFormat.Group ? WireType.StartGroup : WireType.String;
             }
 
-            switch (Helpers.GetTypeCode(type))
+            switch (Helpers.GetTypeCode(type, true))
```

Comment:
interesting that this happens after the enum/model checks; this makes me wonder how `<SomeEnum?>` and `<SomeValueTypeCustomModel?>` behave - I'll need to look locally to try to assess the impact of this proposed change